src/charm.py

```diff
@@ -1,103 +1,217 @@
+            "CONFIG_ROOT": "/configs",
+            "TEMPORAL_HOST": self.config["temporal-host"],
+            "WORKER_LOGS_STORAGE_TYPE": self.config["storage-type"].value,
+            "WORKER_STATE_STORAGE_TYPE": self.config["storage-type"].value,
+            "STORAGE_TYPE": self.config["storage-type"].value,
+            "STORAGE_BUCKET_ACTIVITY_PAYLOAD": "payload-storage",
+            "STORAGE_BUCKET_LOG": self.config["storage-bucket-log"],
+            "STORAGE_BUCKET_STATE": self.config["storage-bucket-state"],
+            "STORAGE_BUCKET_WORKLOAD_OUTPUT": self.config["storage-bucket-workload-output"],
+            "CONFIGS_DATABASE_MINIMUM_FLYWAY_MIGRATION_VERSION": "0.40.23.002",
+            "JOBS_DATABASE_MINIMUM_FLYWAY_MIGRATION_VERSION": "0.40.26.001",
+            "MICRONAUT_ENVIRONMENTS": "control-plane",
+            "WORKERS_MICRONAUT_ENVIRONMENTS": "control-plane",
+            "CRON_MICRONAUT_ENVIRONMENTS": "control-plane",
+            "INTERNAL_API_HOST": f"localhost:{INTERNAL_API_PORT}",
+            "LOG_LEVEL": self.config["log-level"].value,
+            "MICROMETER_METRICS_ENABLED": "false",
+            "KEYCLOAK_INTERNAL_HOST": "localhost",
+            "KEYCLOAK_DATABASE_URL": db_url + "?currentSchema=keycloak",
+            "WEBAPP_URL": "airbyte-ui-k8s:8080",
+            "WORKER_ENVIRONMENT": "kubernetes",
+            "WORKSPACE_DOCKER_MOUNT": "airbyte_workspace",
+            "SECRET_PERSISTENCE": "TESTING_CONFIG_DB_TABLE",
+            "CONNECTOR_BUILDER_SERVER_API_HOST": f"localhost:{CONNECTOR_BUILDER_SERVER_API_PORT}",
+            "S3_LOG_BUCKET_REGION": "",
+            "MINIO_ENDPOINT": "http://minio:9000",
+            "S3_LOG_BUCKET": "airbyte-dev-logs",
+            "S3_PATH_STYLE_ACCESS": "true",
+            "SHOULD_RUN_NOTIFY_WORKFLOWS": "true",
+            "STATE_STORAGE_MINIO_ACCESS_KEY": "minio",
+            "STATE_STORAGE_MINIO_SECRET_ACCESS_KEY": "minio123",
+            "STATE_STORAGE_MINIO_BUCKET_NAME": "state-storage",
+            "STATE_STORAGE_MINIO_ENDPOINT": "http://minio:9000",
+            "AIRBYTE_API_HOST": "localhost:8006",
+            "CONNECTOR_BUILDER_API_URL": "/connector-builder-api",
+            "WORKLOAD_API_HOST": "localhost:8007",
+            "WORKLOAD_API_URL": "localhost:8007",
+            "TEMPORAL_WORKER_PORTS": "9001,9002,9003,9004,9005,9006,9007,9008,9009,9010,9011,9012,9013,9014,9015,9016,9017,9018,9019,9020,9021,9022,9023,9024,9025,9026,9027,9028,9029,9030",
+            "AWS_ACCESS_KEY_ID": "minio",
+            "AWS_SECRET_ACCESS_KEY": "minio123",
+            "JOB_KUBE_SERVICEACCOUNT": "airbyte-k8s",
+            "JOB_KUBE_NAMESPACE": "dev-airbyte",
+            "RUNNING_TTL_MINUTES": self.config["pod-running-ttl-minutes"],
+            "SUCCEEDED_TTL_MINUTES": self.config["pod-successful-ttl-minutes"],
+            "UNSUCCESSFUL_TTL_MINUTES": self.config["pod-unsuccessful-ttl-minutes"],
+            "HTTP_PROXY": "http://squid.internal:3128",
```

Comment:
i guess you are planning to get this from the model config right? same for the noproxy below and the canonical specific stuff.. and also since these repeat, maybe some variables could help..

Reply:
Yup, will be extracting these from model config into variables and reusing them.